snapshot-cross-account-replicator/functions/lambda.py

```diff
@@ -211,31 +192,34 @@ def cleanup_snapshots(event, context):
 
             for page in page_iterator:
                 for snapshot in page['DBClusterSnapshots']:
-                    delete_old_snapshot(rds, snapshot)
+                    delete_old_snapshot(rds, snapshot, older_than)
         else:
             paginator = rds.get_paginator('describe_db_snapshots')
             page_iterator = paginator.paginate(
                 DBInstanceIdentifier=instance, SnapshotType='manual')
 
             for page in page_iterator:
                 for snapshot in page['DBSnapshots']:
-                    delete_old_snapshot(rds, snapshot)
+                    delete_old_snapshot(rds, snapshot, older_than)
+
+def cleanup_intermediate_snapshots(event, context):
+    """Lambda entry point for the cleanup intermediate snapshots"""
 
+    cleanup_snapshots(retention_period + safe_period)
```

Comment:
Afaik this means that intermediate snapshots are no longer cleaned up immediately, but actually are retained longer (2 days) than the replicated snapshots.

This can have cost impacts, essentially storing snapshots twice.

Instead I'd remove snapshots immediately after the `safe_period` (*), not `retention_period + safe_period`.

(*) 1 day safe period should also be quite enough, no?